### PR TITLE
fix deprications and add unit of measurement to battery_level sensor

### PR DESCRIPTION
**Pull request by ChristophCaina**
* fix deprication warnings for HA 2024.1 #
* fix missing Unit_of_Measurement for sensor Battery_Level

at least, partially fixed issues:

#45
#51 
#44
#36
#31

### CODE REVIEW

**Comment by ChristophCaina** [comment]
[c143558](https://github.com/benleb/sureha/pull/53/commits/c143558f62e99b9b2f108da43d90945b1c86d223) - adds german translation

**Review by paulcombuse — APPROVED** [review]
Changes correct due to deprecation:
[VOLUME_MILLILITERS substituted by UnitOfVolume.MILLILITERS](https://github.com/home-assistant/core/blob/f793fbe49230c24c3d9f0b32ccceda839968907c/homeassistant/const.py#L1016)
[MASS_GRAMS substituted by UnitOfMass.GRAMS](https://github.com/home-assistant/core/blob/f793fbe49230c24c3d9f0b32ccceda839968907c/homeassistant/const.py#L1079)
[DEVICE_CLASS_BATTERY substituted by SensorDeviceClass.BATTERY](https://github.com/home-assistant/core/blob/f793fbe49230c24c3d9f0b32ccceda839968907c/homeassistant/const.py#L325)
[DEVICE_CLASS_CONNECTIVITY substituted by BinarySensorDeviceClass.CONNECTIVITY](https://github.com/home-assistant/core/blob/cfd1f7809f6f6dc3a40682719036f4479c28bbb3/homeassistant/components/binary_sensor/__init__.py#L150)
[DEVICE_CLASS_PRESENCE substituted by BinarySensorDeviceClass.PRESENCE](https://github.com/home-assistant/core/blob/cfd1f7809f6f6dc3a40682719036f4479c28bbb3/homeassistant/components/binary_sensor/__init__.py#L192)

**Comment by paulcombuse** [comment]
@benleb please create a pull request with the changes of @ChristophCaina - many thanks! 🍺

**Comment by blair287** [comment]
> @benleb please create a pull request with the changes of @ChristophCaina - many thanks! 🍺

Don't think he is around anymore so not sure we will be getting anymore updates.

**Comment by pdwonline** [comment]
Anyone willing to share a fork for this as it seems stale?

**Comment by ChristophCaina** [comment]
I mean, I can revert other changes I've made in my fork - and provide the fork "as is" - but I cannot provide more support on the component.

For the time being, you could manually put the changes into the folder of your custom_component.

**Comment by benleb** [comment]
please ping me when you are "ready" :) 👍

**Comment by ChristophCaina** [comment]
Hi, ready for what? ^^
I can submit a pr based in the current Progress - I've created new File struckture and Made it ready for releases ...
But this pr should also Contain all relevant Changes

Are you in discord?

File: README.md
Comment (benleb) [inline README.md:1]: I cant merge it with this 😅

**Comment by benleb** [comment]
ready for merge :D For example the readme file (see the comment) seems not ready to be merged (and maybe some other files too?) :)

File: README.md
Comment (ChristophCaina) [reply README.md:1]: lol... ok.
I honestly thought, I removed that already...
Seems that it was re-added with recent changes.

Sorry

**Comment by ChristophCaina** [comment]
I'll create a new Pull-request... sorry...

**Comment by benleb** [comment]
all good, np :) 👍  thanks for your work!

**Pull request closed without merging**